### PR TITLE
refact: restrict image push with KYC level

### LINKED ISSUE #1434: Disable task push if the buyer does not have KYC3

### PR DESCRIPTION
**Pull request by antmat**
This PR introduces acl which checks KYC level and allow only users of specific level to push custom inages on worker. Also it's made configurable and it is reused in whitelist configuration.
closes #1434

### CODE REVIEW

File: insonmnia/worker/server.go [hunk 1]
```diff
@@ -607,7 +610,7 @@ func (m *Worker) taskAllowed(ctx context.Context, request *pb.StartTaskRequest)
 	if err != nil {
 		return false, nil, err
 	}
-	if level <= pb.IdentityLevel_REGISTERED {
+	if level <= m.cfg.Whitelist.PrivilegedIdentityLevel {
```
Comment (sshaman1101) [inline insonmnia/worker/server.go]: Seems that only users which level is `less` than required should be checked with the allowed list.

File: insonmnia/worker/acl_test.go [hunk 2]
```diff
@@ -140,3 +141,27 @@ func TestMultiAuth(t *testing.T) {
 	err = mul.Authorize(context.Background(), nil)
 	assert.Error(t, err)
 }
+
+func TestKycAuthorization(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	identifiedMock := NewMockkycFetcher(ctrl)
+	identifiedMock.EXPECT().GetProfileLevel(gomock.Any(), addr).AnyTimes().Return(sonm.IdentityLevel_IDENTIFIED, nil)
+
+	// Excatly same level
```
Comment (3Hren) [inline insonmnia/worker/acl_test.go]: Typo.

File: insonmnia/worker/config.go [hunk 3]
```diff
@@ -63,6 +66,10 @@ func NewConfig(path string) (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
+	// TODO: drop in next major version
```
Comment (3Hren) [inline insonmnia/worker/config.go]: Add tracking issue.

File: insonmnia/worker/config.go [hunk 4]
```diff
@@ -63,6 +66,10 @@ func NewConfig(path string) (*Config, error) {
...
+	if *cfg.Whitelist.Enabled == false {
```
Comment (3Hren) [inline insonmnia/worker/config.go]: `if !*cfg.Whitelist.Enabled`

**Pull request merged**